qadence_libs/qinfo_tools/qng.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+from typing import Callable
+
+import torch
+from qadence import QuantumCircuit
+from torch.optim.optimizer import Optimizer, required
+
+from qadence_libs.qinfo_tools.qfi import get_quantum_fisher, get_quantum_fisher_spsa
+
+
+class QNG(Optimizer):
+    """Implements the Quantum Natural Gradient Algorithm.
+
+    WARNING: This class implements the exact QNG optimizer, which is very inefficient
+    both in time and memory as it calculates the exact Quantum Fisher Information of
+    the circuit at every iteration. Therefore, it is not meant to be run with medium
+    to large circuits. Other approximations such as the QNG-SPSA optimizer are much
+    more efficient while retaining good performance.
+    """
+
+    def __init__(
+        self,
+        params: tuple | torch.Tensor,
+        circuit: QuantumCircuit = required,
+        lr: float = required,
+        beta: float = 10e-3,
+    ):
+        """
+        Args:
+
+            params (tuple | torch.Tensor): Variational parameters to be updated
+            circuit (QuantumCircuit): Quantum circuit. Needed to compute the QFI matrix
+            lr (float): Learning rate.
+            beta (float):
+                Shift applied to the QFI matrix before inversion to ensure numerical stability.
+                Defaults to 10e-3.
+        """
+
+        if not 0.0 <= lr:
```

Comment:
Isn't it simpler to do:

```suggestion
        if 0.0 > lr:
```